src/algebra/module.lean

```diff
@@ -179,61 +208,70 @@ instance semiring.to_semimodule [semiring R] : semimodule R R :=
   one_smul := one_mul,
   zero_smul := zero_mul,
   smul_zero := mul_zero }
+end
 
 @[simp] lemma smul_eq_mul [semiring R] {a a' : R} : a • a' = a * a' := rfl
 
-instance ring.to_module [ring R] : module R R :=
-{ to_semimodule := semiring.to_semimodule }
-
-/-- A ring homomorphism `f : R →+* S` defines a semimodule structure by `r • x = f r * x`. -/
-def ring_hom.to_semimodule [semiring R] [semiring S] (f : R →+* S) : semimodule R S :=
+/-- A ring homomorphism `f : R →+* M` defines a module structure by `r • x = f r * x`. -/
+def ring_hom.to_semimodule [semiring R] [ring M] (f : R →+* M) : semimodule R M :=
+semimodule.of_core
```

Comment:
This change makes `ring_hom.to_semimodule` require `[ring M]` instead of `semiring`. Is there any reason?

Reply:
Mistake on my side. There were two proofs, one for semirings and one for rings, and I kept the bad one. Corrected.